Yoast/wordpress-seo
# Mark translations that are only for screen readers

## Context


* Improve translator context by adding a `Hidden accessibility text.` comment whenever translations are _only_ used in either screen reader texts or aria labels. To be clear: this means searching for the same translation in the same text domain, if it is used in any other context than screen reader/aria label that means no comment should be there. This is due comments appearing above _all_ the references.

## Summary


This PR can be summarized in the following changelog entry:

* Adds mark above translations that are used only for screen readers.
* [@yoast/components 0.0.1] Adds mark above translations that are used only for screen readers.
* [@yoast/search-metadata-previews 0.0.1] Adds mark above translations that are used only for screen readers.
* [@yoast/components 0.0.1] Removes translation around variable in the HelpIcon component. The `linkText` prop is expected to already be translated. 
* Fixes a bug where the `aria-label` text of the `Learn more` links in the Site features settings where not translated.

## Relevant technical choices:

* I went with the WP version: `/* translators: Hidden accessibility text. */`.
* Out of scope changes:
  * Ignoring PHPCS output not escaped instance where I saw it was all properly escaped above.
  * Removing the `__()` around the `linkText` prop of the HelpIcon component.
    * AFAIK, the retrieving of translations won't follow variables. Coincidentally, the implementers of this variable seem to all already call `__()` on the string itself.
  * Fixing an untranslated aria-label in settings -> site features.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* It would be nice to have a search yourself, to see if I missed any instances?
  * Note that I intentionally ignored where the variable was used in more than just the screen reader, or in apps, or deprecated. Or when the same translation was used in a non screen reader capacity elsewhere.
* The comments should show up in the `.pot` file after running the i18n grunt task.
* Not really testable until translations are there, but when they are there:
  * Switch site away from English
  * Go to Yoast settings -> Site features
  * Verify the `aria-label` of the `Learn more` links are translated

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Not really testable until translations are there, but when they are there:
  * Switch site away from English
  * Go to Yoast settings -> Site features
  * Verify the `aria-label` of the `Learn more` links are translated
* The rest is not testable, since its only code comments.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* Should only be comments in the code, without any consequences in production.

## UI changes


## Other environments


## Documentation


## Quality assurance


## Innovation


Fixes https://github.com/Yoast/wordpress-seo/issues/19854
